concepts/05 UI Components/TreeList/10 Columns/50 Adaptability/15 Customize Adaptive Detail Row.md

```diff
@@ -0,0 +1,101 @@
+Adaptive detail row contains the [Form](/Documentation/ApiReference/UI_Components/dxForm/) UI component. You can implement the [onAdaptiveDetailRowPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onAdaptiveDetailRowPreparing) handler to customize the Form: change its properties in the **formOptions** attribute of the function's argument. For example, the following code marks the form item whose data field is *"OrderID"* as required:
```

Comment:
```suggestion
Adaptive detail rows contain the [Form](/Documentation/ApiReference/UI_Components/dxForm/) UI component. You can implement the [onAdaptiveDetailRowPreparing](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#onAdaptiveDetailRowPreparing) handler to customize the Form: change its properties in the **formOptions** attribute of the function's argument. For example, the following code marks the form item whose data field is set to *"OrderID"* as required:
```

Reply:
The edited sentence - "For example, the following code marks the form item whose data field is **_set to_** *"OrderID"* as required" - means this:
```
{
  dataFieldName: "OrderID"
}
```
But it should mean this:
```
{
  OrderID: someOrderIdValue
}
```
Can the sentence be "For example, the following code marks the form's *"OrderID"* data field as required"?

Reply:
Marks it as a field that must be filled, e.g.:
![image](https://user-images.githubusercontent.com/66619750/114987696-1f2d0e80-9e9e-11eb-9cb1-b93e4ad4a575.png)

Reply:
ok, that's fine then